lang/syn/src/parser/accounts/mod.rs

```diff
@@ -178,6 +178,27 @@ fn constraints_cross_checks(fields: &[AccountField]) -> ParseResult<()> {
                         ));
                     }
                 }
+
+                // Make sure initialiazed token accounts are always declared after their corresponding mint.
+                InitKind::Mint { .. } => {
+                    if init_fields.iter().enumerate().any(|(f_pos, f)| {
+                        match &f.constraints.init.as_ref().unwrap().kind {
+                            InitKind::Token { mint, .. }
+                            | InitKind::AssociatedToken { mint, .. } => {
+                                mint.to_token_stream()
```

Comment:
I think we should check equality instead of `starts_with` here. This will incorrectly throw error in a case like this:

```rs
#[account(
    init,
    payer = payer,
    mint::decimals = 0,
    mint::authority = payer,
)]
pub mint2: Account<'info, Mint>,

#[account(
    init,
    payer = payer,
    token::mint = mint2,
    token::authority = payer,
)]
pub token: Account<'info, TokenAccount>,

#[account(
    init,
    payer = payer,
    mint::decimals = 0,
    mint::authority = payer,
)]
pub mint: Account<'info, Mint>,
```

Reply:
Indeed!